### ISSUE TITLE
KDL 2.0: multiline comments in more places

### ISSUE THREAD

**Issue by zkat** [kdl-org/kdl#342]
stuff like `foo/*bar*/baz 1` should be legal (resolving to `foobaz 1` in this case)

Ref: #270

**Comment by tabatkins** [comment]
Wait, you actually want the comments to be able to divide individual meaningful tokens? That seems fraught, I think.

**Comment by zkat** [comment]
yeah? that's what the original request was, no?

**Comment by LemmaEOF** [comment]
I kinda feel like this should resolve to `foo baz 1` as comments are considered whitespace

**Comment by dezren39** [comment]
i think this should be legal, i don't know what it should do.

**Comment by pingiun** [comment]
> I kinda feel like this should resolve to `foo baz 1` as comments are considered whitespace

100% agree. If you treat comments as whitespace that allows you to easily create tokenizing parsers. Treating comments as nothing space makes it more confusing I think.

```kdl
foo/*bar */baz 1
```
would be different from
```kdl
foo/*bar*/ baz 1
```

I feel like it makes sense for them to be the same

**Comment by tabatkins** [comment]
Note for others: `foo/**/bar` is *already* valid KDL, and *is* equivalent to `foo bar`. Right now, block comments are treated as a type of whitespace; see the `ws := ...` line in [the grammar]https://github.com/kdl-org/kdl/blob/main/SPEC.md).

The request here from #270 is to allow comments in some places whitespace isn't currently allowed, but still between "tokens". So `(foo/**/)node` or `node foo/**/=2`. `foo/**/bar` or `2/**/.5` would continue to be equivalent to `foo bar` and `2 .5`; `2./**/5` would be invalid since `2.` isn't a valid number.

Given that #340 is asking for allowing whitespace between more tokens (around the `=`), we might be able to resolve this solely by just allowing whitespace around the ident in a tag, so `( foo )node` is allowed. Then comments would be valid *by default* there.

If we did that, then I think there's only *one* remaining spot where two adjacent tokens can't have whitespace between them, and that's the tag and node, like `(foo) node` is illegal today. I'm weakly against allowing whitespace there (very weakly), but would be fine with explicitly allowing block comments there regardless.

**Comment by LemmaEOF** [comment]
Yeah, `( foo )node` should definitely be allowed. I'm somewhat in favor of `(foo) node` as well but I understand why separating a tag from its element could be Semantically Frustrating

**Comment by zkat** [comment]
Closing in favor of https://github.com/kdl-org/kdl/issues/355 for clarity
